### ISSUE TITLE
easier distro addition

### ISSUE THREAD

**Issue by igel-kun** [GameServerManagers/LinuxGSM#2370]
## User Story
I've been trying to add (dependency) support for the distro I'm running (Gentoo) but it's somewhat hard to parse through all the distro-check code... can't we have an infrastructure like, a directory `distros` containing only distro-specific scripts. I would love to be able to just take the script for debian and change 2 functions: "test_distro()" returning 1 if the current distro matches the script (and requirements are there, such as "rpm is installed") and "test_dependency()" returning 1 if a given program is installed or not. DONE.

I don't feel comfortable enough with your code to create such an infrastructure myself, so this is a 
feature request :)

cheers

**Comment by dgibbs64** [comment]
The only real difference between the supported distros is the dependencies. Other than that very little specific code is required. The dependency module is quite complex because of all the variations required for all the various distros and game servers. Even similar distro derivatives have differences that can be problematic. I don't think a refactor in the check dependecy module is currently a priority and dont see much benefit currently. However, adding support for Gentoo would be good.
